cost/flexera/cco/scheduled_reports/scheduled_report.pt

```diff
@@ -543,13 +602,96 @@ script "js_collated_data", type: "javascript" do
       month = timestamp.toISOString().split('-')[1]
       stringMonth = ds_data_tables['months_short'][Number(timestamp.toISOString().split('-')[1])]
       stringDay = stringMonth + "-" + day
+      // Get days in month for yearMonth
+      var daysinTimeUnit = new Date(year, month, 0).getDate()
 
       category_id = ds_graph_dimension['id']
       if (category_id == "billing_center_id") { category_id = "billing_center_name" }
       category = row['dimensions'][category_id]
 
-      if (param_billing_term == 'Day')   { timeUnit = timestamp.toISOString().split('T')[0] }
-      if (param_billing_term == 'Month') { timeUnit = [year, month].join('-') }
+      if (param_billing_term == 'Day')   {
+        timeUnit = timestamp.toISOString().split('T')[0]
+        // Override daysinTimeUnit for daily average calculation
+        // At param_billing_term Day, this Average is not useful and will match the cost
+        daysinTimeUnit = 1
+      }
+      if (param_billing_term == 'Month') {
+        timeUnit = [year, month].join('-')
+      }
+      // For this graph dimension value, if not yet defined create an empty object
+      if (typeof previousTimePeriod[category] == "undefined") {
+        previousTimePeriod[category] = {}
+      }
+
+      // Calculate daily average cost
+      var dailyAverageCost = row['metrics'][metric] / daysinTimeUnit
+
+      // Check if we can calculate Percent Change from previous time period
+      var percentChange = null
+      var percentChangeDailyAverageCost = null
+      var isPercentChangeAboveThreshold = false
+      if (typeof previousTimePeriod[category]["row"] == "object") {
+        var previousCost = previousTimePeriod[category]["row"]['metrics'][metric]
+        var previousDailyAverageCost = previousTimePeriod[category]['dailyAverageCost']
+        if (previousCost > 0) {
+          var currentCost = row['metrics'][metric]
+          percentChange = Number((((currentCost - previousCost) / previousCost) * 100 ).toFixed(3))
+          percentChangeDailyAverageCost = Number((((dailyAverageCost - previousDailyAverageCost) / previousDailyAverageCost) * 100 ).toFixed(3))
+        } else {
+          percentChange = Number(row['metrics'][metric].toFixed(3))
+          percentChangeDailyAverageCost = Number(dailyAverageCost.toFixed(3))
+        }
+        // See if the percent change is above the threshold
+        // Take the absolute value of percentChange so we can compare it to the threshold.  Otherwise, by default and most use-cases any negative percent change would be below the threshold which would trigger a report which is not generally needed for downward change
+        if (param_percent_change_alert_metric == "Percent Change from Previous Time Period Daily Average") {
+          // Switch to handle param_percent_change_alert_direction
+          if (param_percent_change_alert_direction == "Increase") {
```

Comment:
Might be able to reduce some of this code here by setting some value to either 1 or -1 depending on whether param_percent_change_alert_direction == "Increase" or not. Would eliminate some redundant code, since you could reduce the number of distinct comparisons by half.

Reply:
Refactored in [5748f63](https://github.com/flexera-public/policy_templates/pull/2485/commits/5748f637c6e0dc9a77bed24f62006df642d39b8c).  Let me know if you think this is the right approach